### PR TITLE
Broadcast sparse dot

### PR DESCRIPTION
**Pull request by twiecki**
Supersedes https://github.com/Theano/Theano/pull/6522

### CODE REVIEW

**Comment by abergeron** [comment]
There is another mix of tabs and spaces on the line below.

File: theano/sparse/tests/test_basic.py [hunk 1]
```diff
@@ -464,6 +464,23 @@ def test_dot(self):
...
+                (tensor.vector()[None, :], SparseType('csr', 'float32')()),
+                (tensor.matrix(), SparseType('csr', 'float32')())]:
+
+            sparse_out = theano.dot(x, y)
+            if isinstance(x, sparse.SparseVariable):
+                x = tensor.matrix()
+            if isinstance(y, sparse.SparseVariable):
+                y = tensor.matrix()
+            dense_out = tensor.dot(x, y)
+	    assert dense_out.broadcastable == sparse_out.broadcastable
```
Comment (abergeron) [inline theano/sparse/tests/test_basic.py]: This starts with a tab

Comment (twiecki) [reply theano/sparse/tests/test_basic.py]: Fixed.

**Comment by twiecki** [comment]
@abergeron Tests passed.

**Pull request merged**